tests/unit/resources/test_compensation.py

```diff
@@ -98,3 +108,143 @@ def test_create_from_spill_string(spillstring):
+    assert all(results == events)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_matching_kwargs(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=open("tests/data/Acea - Novocyte.fcs", "rb"),
+    )
+    events = fcs_file.get_events(inplace=True, testKwarg="foo")
+    assert fcs_file._events_kwargs == {"testKwarg": "foo"}
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24), index=ix, columns=ix)
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file, testKwarg="foo")
+
+    # Then: events should be scaled
+    assert all(results == events)
+    assert (
+        responses.assert_call_count(
+            f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs?testKwarg=foo",
+            1,
+        )
+        is True
+    )
+
+
+@responses.activate
+def test_apply_scale_with_subset_of_values(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+    test_col = events["FSC-H"]
+
+    ix = list(events.columns)[0:10]
+    compensation.dataframe = DataFrame(identity(10, int), index=ix, columns=ix)
+    compensation.dataframe = compensation.dataframe * 4
+
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file, inplace=False)
+
+    # Then: events should be scaled
+    assert all(results["FSC-H"] != test_col)
+
+
+@responses.activate
+def test_apply_scale_compensates_values(client, ENDPOINT_BASE, compensation, fcs_file):
+    # Given: a Compensation with channels as a subset of the FcsFile events
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+    test_col = events["FSC-H"]
+
+    ix = list(events.columns)
+    compensation.dataframe = DataFrame(identity(24, int), index=ix, columns=ix)
+    compensation.dataframe = compensation.dataframe * 4  # modify the matrix
+    compensation.channels = ix
+
+    # When: a Compensation is applied
+    results = compensation.apply(fcs_file, inplace=False)
+
+    # Then: events should be scaled
+    assert all(results["FSC-H"] != test_col)
```

Comment:
This test should be a bit more rigorous.

I'd upload the Novocyte file to CellEngine, then go to Export Populations and pick "TSV" format, compensated, file-internal comp. You'll get a TSV file that you can use as the approximate truth values (likely within 1e-5). Then in this test you can create the comp from the file's spill string and assert on the values of the first row of the dataframe.